numpy/random/mtrand/mtrand.pyx

```diff
@@ -4376,12 +4381,20 @@ cdef class RandomState:
         # not zero. We continue to use the SVD rather than Cholesky in
         # order to preserve current outputs. Note that symmetry has not
         # been checked.
+
         (u, s, v) = svd(cov)
-        neg = (np.sum(u.T * v, axis=1) < 0) & (s > 0)
+
+        neg = (np.sum(u.T * v, axis=1) < 0) & (s > 0) & ~np.isclose(s, 0,
+                                                                    atol=tol)
```

Comment:
Yeah, I have added a symmetry check.

We are just checking the covariance matrix is almost PSD.